plugins/query-insights/src/main/java/org/opensearch/plugin/insights/rules/model/SearchQueryLatencyRecord.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.insights.rules.model;
+
+import org.opensearch.action.search.SearchType;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * The Latency record stored in the Query Insight Framework
+ *
+ * @opensearch.internal
+ */
+public final class SearchQueryLatencyRecord extends SearchQueryRecord<Long> {
```

Comment:
I don't quite get the purpose of having different type of records when it all can be part of one record. 
Like if there is a record for a query provided by this framework and I get measurements associated with it like latency, cpu, heap etc as part of one record, that would be ideal.

Reply:
Thanks for the input! Refactored the code to make the SearchQueryRecord more generic. In that way the SearchQueryLatencyRecord will not be needed anymore.